File: android/app/src/main/java/com/example/socialfeed/data/api/ApiModels.kt
package com.example.socialfeed.data.api

import com.google.gson.annotations.SerializedName

// Auth
data class LoginRequest(val username: String, val password: String)
data class RegisterRequest(val username: String, val password: String)
data class AuthResponse(val token: String, val user: ApiUser)

// User
data class ApiUser(
    val id: Int,
    val username: String,
    val avatarUrl: String?,
    @SerializedName("_count") val count: UserCount? = null
)

data class UserCount(val posts: Int = 0)

data class UpdateUserRequest(val username: String? = null, val avatarUrl: String? = null)

// Post
data class PostsResponse(val posts: List<ApiPost>)

data class ApiPost(
    val id: Int,
    val text: String,
    val imageUrl: String?,
    val createdAt: String,
    val author: ApiUser,
    @SerializedName("_count") val count: PostCount? = null,
    val liked: Boolean? = null
)

data class PostCount(val likes: Int = 0, val comments: Int = 0)

data class CreatePostRequest(val text: String, val imageUrl: String? = null)

// Like
data class LikeResponse(val liked: Boolean, val likesCount: Int)

// Comment
data class CommentsResponse(val comments: List<ApiComment>)

data class ApiComment(
    val id: Int,
    val text: String,
    val createdAt: String,
    val author: ApiUser
)

data class CreateCommentRequest(val text: String)

File: ios/SocialFeed/Models/APIModels.swift
import Foundation

// MARK: - Auth
struct AuthResponse: Codable {
    let token: String
    let user: APIUser
}

struct LoginRequest: Codable {
    let username: String
    let password: String
}

// MARK: - User
struct APIUser: Codable, Identifiable {
    let id: Int
    let username: String
    let avatarUrl: String?
    let _count: UserCount?
    
    struct UserCount: Codable {
        let posts: Int?
    }
}

// MARK: - Post
struct APIPost: Codable, Identifiable {
    let id: Int
    let text: String
    let imageUrl: String?
    let createdAt: String
    let author: APIUser
    let _count: PostCount?
    
    struct PostCount: Codable {
        let likes: Int?
        let comments: Int?
    }
    
    var createdAtDate: Date {
        let f = ISO8601DateFormatter()
        f.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        return f.date(from: createdAt) ?? Date()
    }
}

struct PostsResponse: Codable {
    let posts: [APIPost]
}

struct CreatePostRequest: Codable {
    let text: String
    let imageUrl: String?
}

// MARK: - Like
struct LikeResponse: Codable {
    let liked: Bool
    let likesCount: Int
}

// MARK: - Comment
struct APIComment: Codable, Identifiable {
    let id: Int
    let text: String
    let createdAt: String
    let author: APIUser
    
    var createdAtDate: Date {
        let f = ISO8601DateFormatter()
        f.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        return f.date(from: createdAt) ?? Date()
    }
}

struct CommentsResponse: Codable {
    let comments: [APIComment]
}

struct CreateCommentRequest: Codable {
    let text: String
}

struct UpdateUserRequest: Codable {
    let username: String?
    let avatarUrl: String?
}

struct MeResponse: Codable {
    let user: APIUser
}

struct UserResponse: Codable {
    let user: APIUser
}

struct PostResponse: Codable {
    let post: APIPost
}

struct CommentResponse: Codable {
    let comment: APIComment
}
